src/modules/tracker.js

```diff
@@ -1228,6 +1229,401 @@ class Tracker {
+
+      this.requests.queue(
+        requestURL,
+        requestMethod,
+        requestBody,
+        networkParameters,
+      );
+      this.requests.send();
+
+      return true;
+    }
+
+    this.requests.send();
+
+    return new Error('parameters are required of type object');
+  }
+
+  /**
+   * Send quiz result click event to API
+   *
+   * @function trackQuizResultClick
+   * @param {object} parameters - Additional parameters to be sent with request
+   * @param {string} parameters.quizId - Quiz Id
+   * @param {string} parameters.quizVersionId - Quiz version Id
+   * @param {string} [parameters.itemId] - Product item unique identifier
+   * @param {string} [parameters.itemName] - Product item name
+   * @param {string} [parameters.section='Products'] - Index section
+   * @param {number} [parameters.resultCount] - Total number of results
+   * @param {number} [parameters.resultPage] - The page of the results
+   * @param {string} [parameters.resultId] - Quiz result identifier (returned in response from Constructor)
+   * @param {number} [parameters.resultPositionOnPage] - Position of clicked item
+   * @param {number} [parameters.numResultsPerPage] - Position of clicked item
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {(true|Error)}
+   * @description User viewed a quiz results page
+   * @example
+   * constructorio.tracker.trackQuizResultClick(
+   *     {
+   *         quizId: 'coffee-quiz',
+   *         quizVersionId: '1231244'
+   *         itemId: '123',
+   *         itemName: 'espresso'
+   *     },
+   * );
+   */
+  // eslint-disable-next-line complexity
+  trackQuizResultClick(parameters, networkParameters = {}) {
+    // Ensure parameters are provided (required)
+    if (parameters && typeof parameters === 'object' && !Array.isArray(parameters)) {
+      const requestPath = `${this.options.serviceUrl}/v2/behavioral_action/quiz_result_click?`;
+      const {
+        quizId,
+        quizVersionId,
+        itemId,
+        itemName,
+        resultCount,
+        resultId,
+        resultPage,
+        numResultsPerPage,
+        resultPositionOnPage,
+        section,
+      } = parameters;
+
+      const queryParams = {};
+      const bodyParams = {};
+
+      // Ensure required parameters provided
+      if (typeof quizId !== 'string') {
+        return new Error('"quizId" is a required parameter of type string');
+      }
+
+      if (typeof quizVersionId !== 'string') {
+        return new Error('"quizVersionId" is a required parameter of type string');
+      }
+
+      if (typeof itemId !== 'string' && typeof itemName !== 'string') {
+        return new Error('"itemId" or "itemName" is a required parameter of type string');
+      }
+
+      bodyParams.quizId = quizId;
+      bodyParams.quizVersionId = quizVersionId;
+
+      if (!helpers.isNil(itemId)) {
```

Comment:
Same as above, do we need isNil?

Also, I think we can get rid of the itemId check like you did for quizId, versionId since we already check it above.